runtime/j9vm/javanextvmi.c

```diff
@@ -114,11 +114,16 @@ JVM_GetExtendedNPEMessage(JNIEnv *env, jthrowable throwableObj)
 		J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
 		char *npeMsg = NULL;
 		GetStackTraceElementUserData userData = {0};
+		UDATA skipHiddenFrames = FALSE;
+		/* If -XX:+ShowHiddenFrames option has not been set, skip hidden method frames */
+		if (J9_ARE_NO_BITS_SET(vm->runtimeFlags, J9_RUNTIME_SHOW_HIDDEN_FRAMES)) {
+			skipHiddenFrames = TRUE;
+		}
 
 		Trc_SC_GetExtendedNPEMessage_Entry2(vmThread, throwableObj);
 		vmFuncs->internalEnterVMFromJNI(vmThread);
 		userData.bytecodeOffset = UDATA_MAX;
-		vmFuncs->iterateStackTrace(vmThread, (j9object_t*)throwableObj, getStackTraceElementIterator, &userData, TRUE);
+		vmFuncs->iterateStackTrace(vmThread, (j9object_t*)throwableObj, getStackTraceElementIterator, &userData, TRUE, skipHiddenFrames);
```

Comment:
`BOOLEAN` should be used for `skipHiddenFrames` and `pruneConstructors` since it only needs to store `TRUE/FALSE`.

Reply:
Personally, I'm happy to move away from the use ot `BOOLEAN` - the new parameter is consistent with the existing code.

Reply:
Perhaps not in this PR, but I would prefer using a 'flags' argument instead of multiple boolean parameters. You can't get the order wrong if you're passing only one (flags) argument.

Reply:
This was discussed earlier in the PR a something to do next time we add a flag here.